include/secp256k1_musig.h

```diff
@@ -204,152 +154,95 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_pubkey_tweak_add(
     const unsigned char *tweak32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5);
 
-/** Initializes a signing session for a signer
+/** Starts a signing session
+ *
+ *  This function derives a secret nonce that will be required for signing and
+ *  creates a public nonce that is intended to be sent to other signers.
  *
- *  Returns: 1: session is successfully initialized
- *           0: session could not be initialized: secret key or secret nonce overflow
+ *  MuSig differs from regular Schnorr signing in that implementers _must_ take
+ *  special care to not reuse a nonce. This can be ensured by following these rules:
+ *
+ *  1. Always provide a unique session_id32. It is a "number used once".
+ *  2. If you already know the signing key, message or combined public key, they
+ *     can be optionally povided to derive the nonce and increase
+ *     misuse-resistance. The extra_input32 argument can be used to provide
+ *     additional data that does not repeat in normal scenarios, such as the
+ *     current time.
```

Comment:
> Should we be cautious about making recommendations on relying on local time ?
> 
> Maybe you should precise it should be an internal clock in the same trust boundary as the signer's memory ? Otherwise you might have ingenious device relying on external NTP servers to sync their local date/time.
> 

> Even further, can we assume an adversary to guess your local device's current time or at least picks an iterable range ?

That's not a problem. The requirement is that `extra_input32` does  not repeat, not that it's unpredictable. 

But yeah, maybe time isn't a good recommendation then and we should recommend a large enough counter. 

Maybe this deserves some more thought in a further PR.

Reply:
This recommendation is most useful if you don't have access to good randomness and therefore provide a counter to `session_id32`. If you had good randomness, then time does not repeat should imply randomness does not repeat and therefore adding time wouldn't really help. But agree that this deserves a better explanation.